### PR TITLE
Switch formatting from black+isort to µfmt (black+µsort)

### PR DESCRIPTION
**Pull request by jreese**
Description:

This switches linting, formatting, and pre-commit checks from using black and isort directly to using [µfmt](https://ufmt.omnilib.dev), which runs [µsort](https://usort.readthedocs.io) and black as an atomic formatting step. This matches what torchvision uses (pytorch/vision#4384).

Covered files were formatted using `./tests/run_code_style.sh fmt` and `pre-commit run` validated the results.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: tests/run_code_style.sh [hunk 1]
```diff
@@ -4,13 +4,11 @@ set -xeu
...
+    ufmt diff .
 elif [ $1 = "fmt" ]; then
-    isort . --settings setup.cfg
-    black . --config pyproject.toml
+    ufmt format .
 elif [ $1 = "mypy" ]; then
     mypy --config-file mypy.ini
 elif [ $1 = "install" ]; then
-    pip install flake8 "black==21.12b0" "isort==5.7.0" "mypy"
+    pip install flake8 "black==21.12b0" "usort==1.0.1" "ufmt==1.3.1.post1" "mypy==0.910"
```
Comment (vfdev-5) [inline tests/run_code_style.sh:13]: Why do you fix mypy to 0.910 ?

**Review by vfdev-5 — COMMENTED** [review]
@jreese thanks for the PR ! Yes, ufmt works well in torchvision and maintainers are rather happy with it. 
I left a comment and let me see what other maintainers think about the change as code formatting could be rather opinionated :)

File: tests/run_code_style.sh (hunk 1 above)
Comment (jreese) [reply tests/run_code_style.sh:13]: It was pinned in `run_code_style.bat` as well, so I just assumed it was intended to have pinned versions for everything. Happy to change it back to unpinned if that's desired.

Comment (vfdev-5) [reply tests/run_code_style.sh:13]: oh, I see, we forgot to unpin it there. Yes, please unpin it.

Comment (jreese) [reply tests/run_code_style.sh]: Done!

File: tests/run_code_style.sh [hunk 2]
```diff
@@ -4,13 +4,11 @@ set -xeu
 
 if [ $1 = "lint" ]; then
     flake8 ignite tests examples --config setup.cfg
-    isort . --check --settings setup.cfg
-    black . --check --config pyproject.toml
+    ufmt diff .
```
Comment (vfdev-5) [inline tests/run_code_style.sh:7]: @jreese i wonder why here you use `ufmt diff .` and in `tests/run_code_style.bat` : `ufmt check .` ?

Comment (jreese) [reply tests/run_code_style.sh:7]: Ah, I originally had it as check in both places to match existing behavior, but switched it to diff when I was working out the correct configuration. I personally think diff is more useful, and would suggest using that in both places, but would defer that decision to your opinion.

Comment (vfdev-5) [reply tests/run_code_style.sh:7]: I see. If they both behave similarly in CI and diff shows the diff. I'm ok with `ufmt diff .` everywhere.

**Comment by jreese** [comment]
Rebased onto latest and unified on using `usort diff`.

**Review by vfdev-5 — APPROVED** [review]
Thanks @jreese !

**Pull request merged**